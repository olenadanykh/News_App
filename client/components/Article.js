/* eslint-disable no-useless-constructor */
/* eslint-disable react/prop-types */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable react/prefer-stateless-function */
/* eslint-disable jsx-a11y/alt-text */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
/* eslint-disable react/no-unused-state */
import React, { Component } from 'react';
import '../style/Article.css';

export default class Article extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { allArticles } = this.props;
    const article = allArticles.map((el, i) => {
      if (!el.title) {
        return (
          <div key={i} className="search-article">
            <div className="search-text">
              <h1>{el.abstract}</h1>
              <h2>{el.lead_paragraph}</h2>
              <button onClick={() => window.open(el.url || el.web_url)}> Read More </button>
            </div>
          </div>
        );
      }
      if (el.multimedia) {
        return (
          <div key={i} className="article">
            <img className="image" src={el.multimedia[0].url} />
            <div className="text">
              <h1>{el.title }</h1>
              <h2>{el.abstract}</h2>
              {/* <p>{el.created_date || el.pub_date}</p> */}
              <button onClick={() => window.open(el.url || el.web_url)}> Read More </button>

            </div>
          </div>
        );
      }
    });
    return (
      <div>
        <ul className="container">
          {article}
        </ul>
      </div>
    );
  }
}
