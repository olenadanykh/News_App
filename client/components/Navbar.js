/* eslint-disable no-useless-constructor */
/* eslint-disable react/prop-types */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable react/prefer-stateless-function */
/* eslint-disable jsx-a11y/alt-text */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
/* eslint-disable react/no-unused-state */
import React, { Component } from 'react';
import '../style/Navbar.css';

export default class Navbar extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      handleClick, handleSearch, handleChange, searchVal,
    } = this.props;
    return (
      <div>
        <div className="navbar">
          <h1 onClick={handleClick} id="home" className="logo"> Top_News</h1>
          <form onSubmit={handleSearch}>
            <input id="input" className="input" placeholder="  enter keyword ..." type="text" value={searchVal} onChange={handleChange} />
          </form>
          <ul>
            <li className="category" onClick={handleClick} id="home">Home</li>
            <li className="category" onClick={handleClick} id="politics">Politics</li>
            <li className="category" onClick={handleClick} id="business">Business</li>
            <li className="category" onClick={handleClick} id="technology">Technology</li>
            <li className="category" onClick={handleClick} id="sports">Sports</li>
            <li className="category" onClick={handleClick} id="world">World</li>
          </ul>
        </div>
      </div>
    );
  }
}
