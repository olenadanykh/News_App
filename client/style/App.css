* {
  padding: 0;
  margin: 0;
  
}
*:focus {
  outline: none;
}

body {
margin: 0;
padding: 0;
}

.app.navbar{
  position: fixed;
  justify-content: space-between;
  

}
.navbar{
  background-color: rgb(3, 18, 51);
  height: 100px;
  width: 100%;
  margin-bottom: 30px;
}

 .article-container{
    position: relative;
    display: flex;
    flex-flow: row wrap;
  /* padding: 20px; */
  /* margin-bottom: 30px; */
}

