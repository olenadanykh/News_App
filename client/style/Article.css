
.article{
  background-color: rgb(246, 244, 244);
  box-shadow: rgb(255, 255, 255);
  box-sizing: border-box;
  padding: 20px;
  margin: 20px;
  flex: 1;
  display: flex;
  flex-direction: row;
} 
.search-article{
  background-color: rgb(246, 244, 244);
  box-shadow: rgb(255, 255, 255);
  box-sizing: border-box;
  padding: 20px;
  margin: 20px;
  flex: 1;
  display: flex;
  flex-direction: row;
}
.article > .image{
  flex: 0.5 0 0%;
}

.article img {
display: block;
padding: 5px;
}
.image{
  width: 100px;
  height: 150px;
}
.text{
  padding-left: 20px;
}
.search-text{
  padding-left: 20px;
}

.article > .text{
  flex: 2 0 0%;

}
.text h1{
font-family:Verdana, Geneva, Tahoma, sans-serif;
  font-size: 20px;
  padding: 10px;
}
.search-text h1{
  font-family:Verdana, Geneva, Tahoma, sans-serif;
  font-size: 20px;
  padding: 10px;
}
.text h2{
  font-size: 16px;
  font-weight: normal;
  font-style: italic;
  padding: 10px;
}
.search-text h2{
  font-size: 16px;
  font-weight: normal;
  font-style: italic;
  padding: 10px;
}
.text button{
  padding: 5px;
  background-color:black;
  color:rgb(253, 253, 253);
  border-radius: 10px;
  width: 150px;
  margin-top: 25px;
  margin-left: 80%;
}
.text button:hover{
  background-color:rgb(253, 253, 253);
  color: black;

}
.search-text button{
  padding: 5px;
  background-color:black;
  color:rgb(253, 253, 253);
  border-radius: 10px;
  width: 150px;
  margin-top: 25px;
  margin-left: 80%;
  
}
.search-text button:hover{
  background-color:rgb(253, 253, 253);
  color: black;

}

