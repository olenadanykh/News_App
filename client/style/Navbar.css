
.logo {
  float: left;
  color:rgb(247, 247, 247);
  padding: 20px;
  margin-right: 80px;
  cursor: pointer;
}

.category {
  display:inline-block;
  margin-top: 10px;
  color:rgb(247, 247, 247);
  padding: 20px;
  cursor: pointer;
}
.category:hover{
  opacity: .3;
}

.input{
  float: right;
  height: 30px;
  width: 20%;
  margin-top: 20px;
  margin-right: 30px;


  

}