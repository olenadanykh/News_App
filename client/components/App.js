/* eslint-disable jsx-a11y/alt-text */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
/* eslint-disable react/no-unused-state */
import React, { Component } from 'react';
import Navbar from './Navbar';
import Article from './Article';
import '../style/App.css';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      allArticles: [],
      searchVal: '',
    };
    this.handleSearch = this.handleSearch.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleClick = this.handleClick.bind(this);
  }

  componentDidMount() {
    fetch('http://localhost:3000/home')
      .then(res => res.json())
      .then((data) => {
        // console.log(data.results);
        this.setState({
          allArticles: data.results,
        });
      })
      .catch(err => console.log(err));
  }

  handleClick(e) {
    e.preventDefault();
    // this.setState({ endpoint: e.target.id });
    const val = e.target.id;
    fetch(`http://localhost:3000/${val}`)
      .then(res => res.json())
      .then((data) => {
        console.log(data.results);
        this.setState({
          allArticles: data.results,
        });
      })
      .catch(err => console.log(err));
  }

  handleChange(event) {
    this.setState({ searchVal: event.target.value });
  }

  handleSearch(event) {
    const { searchVal } = this.state;
    event.preventDefault();
    fetch(`https://api.nytimes.com/svc/search/v2/articlesearch.json?q=[${searchVal}]&api-key=ZR52hAUWcADFYvAQCDJfJghhtwRxe4Dm`)
      .then(res => res.json())
      .then((data) => {
        console.log(data);
        this.setState({
          allArticles: data.response.docs,
        });
      })
      .catch(err => console.log(err));
  }

  render() {
    const { allArticles, searchVal, endpoint } = this.state;
    const { handleClick, handleSearch, handleChange } = this;
    return (
      <div className="app">
        {/* <div className="navbar"> */}
        <Navbar
          className="navbar"
          handleClick={handleClick}
          handleSearch={handleSearch}
          handleChange={handleChange}
          searchVal={searchVal}
        />

        {/* </div> */}
        <div className="article-container">
          <Article
            allArticles={allArticles}
          />

        </div>
      </div>
    );
  }
}
