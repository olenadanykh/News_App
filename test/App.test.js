/* eslint-disable react/jsx-filename-extension */

import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../client/components/App';
import Navbar from '../client/components/Navbar';
import Article from '../client/components/Article';


configure({ adapter: new Adapter() });

describe('Test App component', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<App />, { disableLifecycleMethods: true });
    console.log(wrapper.debug());
  });

  it('renders without crashing', () => {
    expect(wrapper.state('searchVal')).toEqual('');
    expect(wrapper.state('allArticles')).toEqual([]);
  });

  it('should not return an error', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should have a child component of type "Navbar" at "index" of 0', () => {
    expect(wrapper.childAt(0).type()).toBe(Navbar);
  });

  it('should have a class of "app"', () => {
    expect(wrapper.hasClass('app')).toBeTruthy();
  });

  it('should be an instance of App', () => {
    expect(wrapper.instance()).toBeInstanceOf(App);
  });

  it('should have a prop of "searchVal" with a value of "news"', () => {
    wrapper.find(Navbar).invoke('handleChange')({ target: { value: 'news' } });
    expect(wrapper.find(Navbar).prop('searchVal')).toBe('news');
  });

  it('should not be falsy because "App" does not return neither null or false', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy();
  });

  it('should return a name with a value of "div"', () => {
    expect(wrapper.name()).toBe('div');
  });

  it('should have a prop "allArticles" with a value of []', () => {
    expect(wrapper.find(Article).props().allArticles).toEqual([]);
  });
});
