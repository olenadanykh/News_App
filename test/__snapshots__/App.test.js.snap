// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component should not return an error 1`] = `ShallowWrapper {}`;
